### ISSUE TITLE
introduce `dict`-like data structures for parameter tables

### ISSUE THREAD

**Issue by sunt05** [UMEP-dev/SUEWS#79]
```
! TODO: all the parameter tables can be stored using a custom dictionary for easier and more consistent access:
```

https://github.com/UMEP-dev/SUEWS/blob/ff9f37b1aeb6dee985bce06abedff4b7bf243efe/SUEWS-SourceCode/src/suews_ctrl_input.f95#L828

**Comment by sunt05** [comment]
related to work in  #34

**Comment by sunt05** [comment]
Potential candidate library: https://github.com/zerothi/fdict

**Comment by sunt05** [comment]
this is now being addressed in #132 by introducing the derived types to allow easier development and maintenance for the future:
- a customised [`f90wrap`](https://github.com/sunt05/f90wrap) package is incorporated;
- top-level physics modules are being modified with new interfaces based on derived types.

**Comment by sunt05** [comment]
This has been addressed in #132 by the introduction of a derived-type-based structural approach.
